applications/OptimizationApplication/custom_conditions/helmholtz_surface_shape_condition.h

```diff
@@ -0,0 +1,238 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 license: OptimizationApplication/license.txt
+//
+//  Main authors:    Reza Najian Asl
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/condition.h"
+#include "includes/model_part.h"
+#include "utilities/integration_utilities.h"
+#include "utilities/geometry_utilities.h"
+#include "optimization_application_variables.h"
+
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/**
+ * @class HelmholtzSurfaceShapeCondition
+ * @ingroup OptimizationApplication
+ * @brief Helmholtz filtering condition for 3D geometries.
+ * @details Implements Sobolev/Helmholtz filter for filtering/regularization. This works for arbitrary geometries in 3D
+ * @author Reza Najian Asl
+ */
+class KRATOS_API(OPTIMIZATION_APPLICATION) HelmholtzSurfaceShapeCondition
+    : public Condition
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Counted pointer of HelmholtzSurfaceShapeCondition
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(HelmholtzSurfaceShapeCondition);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    HelmholtzSurfaceShapeCondition(IndexType NewId, GeometryType::Pointer pGeometry);
+    HelmholtzSurfaceShapeCondition(IndexType NewId, GeometryType::Pointer pGeometry,  PropertiesType::Pointer pProperties);
```

Comment:
```suggestion
    HelmholtzSurfaceShapeCondition(
    IndexType NewId, 
    GeometryType::Pointer pGeometry);
    
    HelmholtzSurfaceShapeCondition(
    IndexType NewId, 
    GeometryType::Pointer pGeometry,  
    PropertiesType::Pointer pProperties);
```

Kratos style, don't commit the suggestion, it kills the tabbing